### ISSUE TITLE
Cluster creation failure with multiple control plane nodes, latest as KUBERNETES_VERSION

### ISSUE THREAD

**Issue by lzhecheng** [kubernetes-sigs/cluster-api-provider-azure#2378]
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
A CI job is creating a capz cluster with 3 control plane nodes, latest as KUBERNETES_VERSION. However, only 1 control plane node is created.

Job config:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml#L643-L652

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/cloud-provider-azure-slow-capz/1535886044187070464
https://storage.googleapis.com/kubernetes-jenkins/logs/cloud-provider-azure-slow-capz/1535886044187070464/build-log.txt
```
Waiting for 3 control plane machine(s), 2 worker machine(s), and  windows machine(s) to become Ready
NAME                              STATUS   ROLES           AGE   VERSION                              INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
capz-swb2s9-control-plane-9gp9f   Ready    control-plane   31m   v1.25.0-alpha.0.930+867b5cc31b376c   10.0.0.4      <none>        Ubuntu 18.04.5 LTS   5.3.0-1034-azure   containerd://1.3.4
capz-swb2s9-md-0-cpm2h            Ready    <none>          28m   v1.25.0-alpha.0.930+867b5cc31b376c   10.1.0.5      <none>        Ubuntu 18.04.5 LTS   5.3.0-1034-azure   containerd://1.3.4
capz-swb2s9-md-0-f57pp            Ready    <none>          29m   v1.25.0-alpha.0.930+867b5cc31b376c   10.1.0.4      <none>        Ubuntu 18.04.5 LTS   5.3.0-1034-azure   containerd://1.3.4
```

**What did you expect to happen:**
3 control plane nodes are created.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
It seems capz is trying to create another AzureMachine resource but it failed.
```
2022-06-12T07:58:28.572746197Z stderr F E0612 07:58:28.572513       1 controller.go:317] controller/azuremachine "msg"="Reconciler error" "error"="failed to reconcile AzureMachine: failed to reconcile AzureMachine service vmextensions: extension state failed. This likely means the Kubernetes node bootstrapping process failed or timed out. Check VM boot diagnostics logs to learn more: failed checking if the operation was complete: failed checking if the operation was complete: Code=\"VMExtensionProvisioningError\" Message=\"VM has reported a failure when processing extension 'CAPZ.Linux.Bootstrapping'. Error message: \\\"Enable failed: failed to execute command: command terminated with exit status=1\\n[stdout]\\n\\n[stderr]\\n\\\"\\r\\n\\r\\nMore information on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot \"" "name"="capz-swb2s9-control-plane-rmvxs" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureMachine" 
```
https://storage.googleapis.com/kubernetes-jenkins/logs/cloud-provider-azure-slow-capz/1535886044187070464/artifacts/management-cluster/capz-control-plane/pods/capz-system_capz-controller-manager-7cc7dcf8f9-2zzm8_7f10082d-a76e-4443-9219-bbd701052cd0/manager/0.log

```
2022-06-12T08:10:04.618425969Z stderr F I0612 08:10:04.618136       1 controller.go:364] controller/kubeadmcontrolplane "msg"="Scaling up control plane" "cluster"="capz-swb2s9" "name"="capz-swb2s9-control-plane" "namespace"="default" "reconciler group"="controlplane.cluster.x-k8s.io" "reconciler kind"="KubeadmControlPlane" "Desired"=3 "Existing"=2
2022-06-12T08:10:04.618547899Z stderr F I0612 08:10:04.618337       1 scale.go:212]  "msg"="Waiting for control plane to pass preflight checks" "cluster-name"="capz-swb2s9" "name"="capz-swb2s9-control-plane" "namespace"="default" "failures"="[machine capz-swb2s9-control-plane-glfxp does not have APIServerPodHealthy condition, machine capz-swb2s9-control-plane-glfxp does not have ControllerManagerPodHealthy condition, machine capz-swb2s9-control-plane-glfxp does not have SchedulerPodHealthy condition, machine capz-swb2s9-control-plane-glfxp does not have EtcdPodHealthy condition, machine capz-swb2s9-control-plane-glfxp does not have EtcdMemberHealthy condition]"
```
https://storage.googleapis.com/kubernetes-jenkins/logs/cloud-provider-azure-slow-capz/1535886044187070464/artifacts/management-cluster/capz-control-plane/pods/capi-kubeadm-control-plane-system_capi-kubeadm-control-plane-controller-manager-d86fc498b-g6kmm_8721e33e-2468-4e7d-8be9-362bfa33099f/manager/0.log
**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by jackfrancis** [comment]
I can repro. Looking through `/var/log/cloud-init-output.log` it appear this may be a problem:

```
[2022-06-17 11:32:46] 2022-06-17 11:32:46 (109 MB/s) - ‘/tmp/k8s-ci/kube-apiserver.tar’ saved [127482880/127482880]
[2022-06-17 11:32:46] 
[2022-06-17 11:32:52] unpacking registry.k8s.io/kube-apiserver-amd64:v1.25.0-alpha.1.29_19c8a2127177b4 (sha256:84addac905501b1f359fa3414c1d1517325a24bbdb1b42afe49b676c65df3cf1)...done
[2022-06-17 11:32:52] ctr: image "k8s.gcr.io/kube-apiserver-amd64:v1.25.0-alpha.1.29_19c8a2127177b4": not found
```

**Comment by jackfrancis** [comment]
@lzhecheng If you update your custom template with the changes in this PR I bet you will be unblocked:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2398

**Comment by jackfrancis** [comment]
@CecileRobertMichon is it possible that @lzhecheng needs to simply update his external cluster template to remove the `ctr images import` and `ctr images tag` statements that we did here?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2356/files#diff-700ec6a406a8b0fe0d9fc91447d53d69822076ace028e6dcaecc3f60a5b4bc58L52

**Comment by CecileRobertMichon** [comment]
Just updating this thread: @jackfrancis and others are investigating, this seems to be an issue with how kubeadm gets the k8s version on the control plane. Removing the statements above won't fix the issue. Stay tuned, looking for a fix.

**Comment by CecileRobertMichon** [comment]
Slack thread here: https://kubernetes.slack.com/archives/CEX9HENG7/p1655475223796599

**Comment by lzhecheng** [comment]
@jackfrancis thank you for effort to investigate! Besides, can we have a CAPZ CI job to verify multiple control-plane-node situation?
